### PR TITLE
move src/config_build.h off AC_CONFIG_HEADERS (#4380)

### PR DESCRIPTION
**Pull request by amiloradovsky**

### CODE REVIEW

**Review by wsnyder — COMMENTED** [review]
Thanks for the fix, but the tests indicate this broke the version handling.

**Comment by amiloradovsky** [comment]
What do dist-vlt-0 and dist-vlt-1 mean?

**Comment by wsnyder** [comment]
dist-vlt-0 and dist-vlt-1are the test groupings to get parallelism, you can ignore them, what matters is the failing test name you can get in the log file

**Comment by amiloradovsky** [comment]
I don't understand why `t_dist_untracked` complains about `config_package.h`, but not any other also generated files

**Review by wsnyder — COMMENTED** [review]
I'm ok if we rename config_build to config_package but how does that help your original problem?

There are other autogenerated things (the HAVE_*'s) in config_build, I believe these will break with this change.

There are some remaining references to config_build in other places, for example the "make clean" rules in src/Makefile.in.

**Comment by amiloradovsky** [comment]
It's not just a rename, it's a split. The auto-generated stuff goes into `config_package.h*` while the handwritten stuff is left in `config_build.h` which is being directly included into the sources. `config_package.h.in` may be rewritten by `autoheader`, and `config_build.h` will be left intact. The original motivation was that `autoreconf` broke everything, later I found that it's enough to simply only do `autoconf`, but still, for the sake of robustness.

Yes, `HAVE_STAT_NSEC`, `HAVE_SYSTEMC`, and `HAVE_COROUTINES` should also be in the auto-generated part, forgot about those.

There should be no remaining mentions of `config_build.h*` as an output. Assuming this search is thorough enough:
```
grep '\<config_build\.h\>' -R ./ --exclude-dir=.git --exclude=\*.h --exclude=\*.cpp --exclude=\*~
```

**Comment by wsnyder** [comment]
Also #define HAVE_CONFIG_BUILD should move over as HAVE_CONFIG_PACKAGE.

**Comment by amiloradovsky** [comment]
`HAVE_CONFIG_BUILD` makes little sense, IMO -- if the header hasn't been generated, the preprocessor won't find it, and the error it'll show isn't any less descriptive than "Something failed during ./configure as config_build.h is incomplete. Perhaps you used autoreconf, don't."

At this point I'm not sure why not just give in to Automake and remove the `config_*.h.in` file completely.
Thus updating the build instructions and CI scripts to use `autoreconf`.

**Comment by wsnyder** [comment]
I don't want to accept changes to the base build process, e.g. using autoreconf at this time.  Please keep HAVE_CONFIG_BUILD.

File: src/config_package.h.in [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+//**********************************************************************
+//**** Configure-discovered library options
+
+// Define if struct stat has st_mtim.tv_nsec (from configure)
+#undef HAVE_STAT_NSEC
+// Define if SystemC found
+// - If defined, the default search path has it, so support is always enabled.
+// - If undef, not system-wide, user can set SYSTEMC_INCLUDE.
+#undef HAVE_SYSTEMC
+#undef HAVE_COROUTINES
```
Comment (wsnyder) [inline src/config_package.h.in:31]:
```suggestion
#undef HAVE_COROUTINES

//**********************************************************************
//**** This file sometimes gets truncated, so check in consumers
#define HAVE_CONFIG_PACKAGE
```

And remove HAVE_CONFIG_BUILD from config_build.h, and replace all usages with HAVE_CONFIG_PACKAGE.

**Review by wsnyder — COMMENTED** [review]
Great, two minor things then think good to merge.

**Pull request merged**